### DISCUSSION TITLE
Printing Color Output

### DISCUSSION THREAD

**Discussion by briandowns** [Jason2605/Dictu#650] [General] [answered]
I haven't dug too far down into this but I was curious how one would print colored output. I've attempted escape codes but not luck.

**Comment by Jason2605** [comment]
Ah yeah good question! Hex escape codes currently aren't being managed so if you try to use an ANSI colour sequence, \xb1 for example, will be the literal string "\xb1" rather than the hex char `'\xb1'`. What we need to do is extend the compiler so that it handles hex values (there will be other sequences as well that aren't being handled)

https://github.com/dictu-lang/Dictu/blob/develop/src/vm/compiler.c#L1020-L1058

**Reply by Jason2605** [reply, marked as answer]
It's a bit nasty and super rough but take a look at the feature/parse_hex branch, that should allow you to use ansi codes for terminal colours

**Comment by briandowns** [comment]
This is great. Thank you. Do we want to support more escape codes?

**Reply by Jason2605** [reply]
We do yeah, as many as possible in reality

**Comment by briandowns** [comment]
Thanks for all of the info. I'll defer to you on how you'd like to proceed. Happy to help with implements.

**Reply by Jason2605** [reply]
Thank you! I think the two to focus on primarily are getting hex and octal numbers into the compiler. There must be a better way than my PoC of parsing them out so I'm gonna do a bit of digging to see if there's a better solution - worst case if there's not the hex solution will work for octal numbers as well

**Reply by briandowns** [reply]
Sounds good. Thanks for all of the quickly replies. 😀

**Comment by briandowns** [comment]
ICYMI:

https://github.com/briandowns/colorize

```sh
dictum install github.com/briandowns/colorize
```
